### PR TITLE
weston: bump to 10.0.0

### PR DESCRIPTION
**Pull request by no92**
This is a tracking PR for updating weston to an up-to-date version.

This PR depends on:
- [x] bringing back and testing Xwayland
- [x] merging managarm/managarm#406
- [x] merging #148, more specifically 11d8b8e
- [x] merging managarm/mlibc#355
- [x] check why `sh` is used in weston-terminal instead of `bash` as before
- [x] fix missing fonts in window titlebars
- [x] update to weston 10 upon release

There might be some more changes needed, I will keep this list updated as we move forward.

### CODE REVIEW

**Comment by no92** [comment]
Updated to weston 10 and cherry-picked the `wayland-protocols` commit this depends on. Ready for review and testing.

The `memfd_create` changes discussed on Discord are not of concern here, as it is only an optional feature that is disabled if `memfd_create` is not present in headers, as it currently is on mlibc master. Stubbing that without implementing it would, however, break weston.

**Comment by no92** [comment]
Dropped the updates to `libdrm` and `wayland-protocols`, as #148 landed them already.

File: patches/weston/0002-Add-xwayland-support-for-managarm.patch [hunk 1]
```diff
@@ -1,30 +0,0 @@
-From 717a54091fe9e2b71d5e00c44b776d8c2c33b0b1 Mon Sep 17 00:00:00 2001
```
Comment (Dennisbonke) [inline patches/weston/0002-Add-xwayland-support-for-managarm.patch:1]: Did you test X based apps without this patch? I'm pretty sure it remains needed and I'm going to double check that myself before we accidentally break Xwayland.

**Review by Dennisbonke — CHANGES_REQUESTED** [review]
No changes needed at the moment, but holding until managarm/mlibc#370 and managarm counterpart are merged and I got a chance at testing Xwayland being operational considering you removed the patch for that.

File: patches/weston/0002-Add-xwayland-support-for-managarm.patch (hunk 1 above)
Comment (no92) [reply patches/weston/0002-Add-xwayland-support-for-managarm.patch:1]: Xwayland seems to work for me. I tested `xclock` and some command-line utilities (I don't remember right now) that operated correctly.

**Comment by avdgrinten** [comment]
Can this be merged now? Since we got the clarification that Xwayland works.

**Comment by Dennisbonke** [comment]
> Can this be merged now? Since we got the clarification that Xwayland works.

I plan on testing this today and merging if it works (which seems to be the case).

**Review by Dennisbonke — CHANGES_REQUESTED** [review]
Holding as during testing it consistently crashes for me with memfd. Also holding on the fact that other packages break with memfd (mesa is one notable contender).

**Comment by no92** [comment]
Dennis' crashes were traced to memfd not setting `_link`, which seems to cause nullptr dereferences in fstatat handling. Apparently, the flag `AT_EMPTY_PATH` is also set in those requests.

**Comment by Dennisbonke** [comment]
On that note, #160 seems to unbreak `mesa`, but without a proper testbed I can't confirm that.

**Comment by Dennisbonke** [comment]
Holding until managarm/managarm#418, managarm/mlibc#370 and #160 are merged

**Review by Dennisbonke — APPROVED** [review]
LGTM now

**Pull request merged**